plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -108,31 +109,44 @@ type resourceRequest struct {
 func getResourceRequest(pod *api.Pod) resourceRequest {
 	result := resourceRequest{}
 	for ix := range pod.Spec.Containers {
-		limits := pod.Spec.Containers[ix].Resources.Limits
-		result.memory += limits.Memory().Value()
-		result.milliCPU += limits.Cpu().MilliValue()
+		requests := pod.Spec.Containers[ix].Resources.Requests
+		result.memory += requests.Memory().Value()
+		result.milliCPU += requests.Cpu().MilliValue()
 	}
 	return result
 }
 
-func CheckPodsExceedingCapacity(pods []*api.Pod, capacity api.ResourceList) (fitting []*api.Pod, notFitting []*api.Pod) {
+func CheckPodsExceedingCapacity(pods []*api.Pod, capacity api.ResourceList, maxPods int64) (fitting []*api.Pod, notFitting []*api.Pod) {
 	totalMilliCPU := capacity.Cpu().MilliValue()
 	totalMemory := capacity.Memory().Value()
 	milliCPURequested := int64(0)
 	memoryRequested := int64(0)
+	// Decide which pods can fit based on CPU/memory requests.
+	var bestEffortPods []*api.Pod
 	for _, pod := range pods {
-		podRequest := getResourceRequest(pod)
-		fitsCPU := totalMilliCPU == 0 || (totalMilliCPU-milliCPURequested) >= podRequest.milliCPU
-		fitsMemory := totalMemory == 0 || (totalMemory-memoryRequested) >= podRequest.memory
-		if !fitsCPU || !fitsMemory {
-			// the pod doesn't fit
-			notFitting = append(notFitting, pod)
-			continue
+		if qos.IsBestEffort(&pod.Spec) {
+			bestEffortPods = append(bestEffortPods, pod)
+		} else {
+			podRequest := getResourceRequest(pod)
+			fitsCPU := totalMilliCPU == 0 || (totalMilliCPU-milliCPURequested) >= podRequest.milliCPU
+			fitsMemory := totalMemory == 0 || (totalMemory-memoryRequested) >= podRequest.memory
+			if !fitsCPU || !fitsMemory {
+				// the pod doesn't fit
+				notFitting = append(notFitting, pod)
+				continue
+			}
+			// the pod fits
+			milliCPURequested += podRequest.milliCPU
+			memoryRequested += podRequest.memory
+			fitting = append(fitting, pod)
 		}
-		// the pod fits
-		milliCPURequested += podRequest.milliCPU
-		memoryRequested += podRequest.memory
-		fitting = append(fitting, pod)
+	}
+	fitting = append(fitting, bestEffortPods...)
```

Comment:
The main issue is the 40 pod limit - if we have 25 best-effort pods and 25 guaranteed pods competing for a node, I think the guaranteed pods should be place onto the node. I think best-effort pods should be "invisible" to guaranteed/burstable pods, so they should not prevent a guaranteed pod from being scheduled. What do you think?